[OSDOCS#18230]: Support TNA with Platform=none (#110815)
openshift/openshift-docs: modules/upi-install-config-local-arbiter-node.adoc

// Module included in the following assemblies:
//
//  installing/installing_bare_metal/upi/installing-bare-metal.adoc

:_mod-docs-content-type: PROCEDURE
[id="upi-install-config-local-arbiter-node_{context}"]

> = Configuring a local arbiter node by using user-provisioned infrastructure

Reviewer: For new content, we do want to include the shortdesc. I think you can get away with using the one you moved to the prereqs here:
```suggestion
= Configuring a local arbiter node by using user-provisioned infrastructure

[role="_abstract"]
To retain high availability (HA) while reducing infrastructure costs for your cluster, you can configure an {product-title} cluster with two control plane nodes and one local arbiter node.
```

Since this is a valid shortdesc that you are already using, you can throw hte tag on too - it shouldn't need a rewrite like the old existing ones earlier.

// Module included in the following assemblies:
//

> //  *installing/installing_bare_metal/ipi/ipi-install-installation-workflow.adoc

Reviewer: This is not where the module is, please ensure you update to use the real paths when you copy from existing modules :)

> :_mod-docs-content-type: PROCEDURE

Reviewer: This is an overloaded module. There is a good chunk of reference content in here ahead of the procedure.

[id="upi-install-config-local-arbiter-node_{context}"]
= Configuring a local arbiter node by using user-provisioned infrastructure

[role="_abstract"]

> You can configure an {product-title} cluster with two control plane nodes and one local arbiter node so as to retain high availability (HA) while reducing infrastructure costs for your cluster.

Reviewer: Suggest flipping this goal-first
```suggestion
To retain high availability (HA) while reducing infrastructure costs for your cluster, you can configure an {product-title} cluster with two control plane nodes and one local arbiter node.
```

Author: @jeana-redhat: Just a question based on this suggestion: To maintain statement consistency for procedures, can we have the standard step structure as **purpose-location-action**? 
For example: 
Instead of "Extract an authentication token to connect to Prometheus by running the following command:"
Make it 
To connect to Prometheus, extract an authentication token by running the following command:

If this looks correct, we need to make this change as one of the tasks for CQA. I found that there is in a way no parallelism maintained throughout guide. 
Let us consider this as a golden chance before migrating to AEM. I know this might increase additional overhead.

Reviewer: Yeah, my training was to always put the goal first, in steps, abstracts, etc. unless it causes a really unwieldy construction. 

As for policy, I know we have that preference for shortdesc blurbs in our guidelines, though I am not sure it is written explicitly. We have some discussion about that [here](https://redhat-internal.slack.com/archives/C04HKR61MN1/p1773181104040269) if you're interested. 

I don't know if we specifically call it out for steps, but I always try to do as you have in your example above and start out "To [accomplish whatever],". I think Ingrid would be a great person to ask about this. I would expect her to know to what degree we encourage this framing officially at a guideline level.

Thanks so much for following through on this! :nerd_face:

> A local arbiter node is a lower-cost, co-located machine that participates in control plane quorum decisions. Unlike a standard control plane node, the arbiter node does not run the full set of control plane services. You can use this configuration to maintain HA in your cluster with only two fully provisioned control plane nodes instead of three.

Reviewer: ```suggestion
A local arbiter node is a lower-cost, colocated machine that participates in control plane quorum decisions. Unlike a standard control plane node, the arbiter node does not run the full set of control plane services. You can use this configuration to maintain HA in your cluster with only two fully provisioned control plane nodes instead of three.
```

https://www.ibm.com/docs/en/ibm-style?topic=grammar-prefixes#hyphens

[IMPORTANT]
====
You can configure a local arbiter node only. Remote arbiter nodes are not supported.
====

To deploy a cluster with two control plane nodes and one local arbiter node, you must define the following nodes in the `install-config.yaml` file:

* 2 control plane nodes
* 1 arbiter node

The arbiter node must meet the following minimum system requirements:

* 2 threads
* 8 GB of RAM
* 50 GB of SSD or equivalent storage

The arbiter node must be located in a network environment with an end-to-end latency of less than 500 milliseconds, including disk I/O. In high-latency environments, you might need to apply the `etcd` slow profile.

The control plane nodes must meet the following minimum system requirements:

* 4 threads
* 16 GB of RAM
* 120 GB of SSD or equivalent storage

> Additionally, the control plane nodes must also have enough storage for the workload.

Reviewer: This is reference content and should be moved to a module about requirements. You can use the module in the IPI version as well.

Author: Created a separate module

.Prerequisites

* You have downloaded {oc-first} and the installation program.
* You have logged into the {oc-first}.

.Procedure

. Edit the `install-config.yaml` file to define the arbiter node alongside control plane nodes.
+
.Example `install-config.yaml` configuration for deploying an arbiter node
[source,yaml]
----
apiVersion: v1
baseDomain: devcluster.openshift.com
compute:
  - architecture: amd64
    hyperthreading: Enabled
    name: worker
    platform: {}
    replicas: 0

# Defines the arbiter machine pool. Configure this field to deploy a cluster with an arbiter node.    
arbiter: 
  architecture: amd64
  hyperthreading: Enabled
  replicas: 1 # Set the `replicas` field to `1` for the arbiter pool. You cannot set this field to a value that is greater than 1.
  name: <Specifies a name for the arbiter machine pool>
  platform:
    baremetal: {}
# Defines the control plane machine pool    
controlPlane:
  architecture: amd64
  hyperthreading: Enabled
  name: master
  platform:
    baremetal: {}
  replicas: 2 # When an arbiter pool is defined, two control plane replicas are valid.

platform:
  none: {}

> ----

Reviewer: We define YAML parameters after the YAML block. Move the parameter explanations outside the block to match our content patterns. For more information and some examples of what these need to look like now that we no longer have callouts, see https://docs.google.com/document/d/1j78SA8Y-ZRlVATbVe6D8bdYj7mQzzhOJtGt1xH_3qa0/
